llvm/include/llvm/CodeGen/GlobalISel/GenericMachineInstrs.h

```diff
@@ -24,6 +24,9 @@ namespace llvm {
 
 /// A base class for all GenericMachineInstrs.
 class GenericMachineInstr : public MachineInstr {
+  constexpr static int PoisonFlags = NoUWrap | NoSWrap | NoUSWrap | IsExact |
```

Comment:
unsigned, but we could probably define this directly in the flags enum

Reply:
I would prefer to not leak GlobalIsel details into MachineInstr.

Reply:
This isn't a GlobalIsel detail. The flags are defined in the base instruction